test/common/stats/stats_matcher_impl_speed_test.cc

```diff
@@ -0,0 +1,73 @@
+// Note: this should be run with --compilation_mode=opt, and would benefit from a
+// quiescent system with disabled cstate power management.
+
+#include "envoy/config/metrics/v3/stats.pb.h"
+#include "envoy/type/matcher/v3/string.pb.h"
+
+#include "source/common/stats/stats_matcher_impl.h"
+#include "test/test_common/utility.h"
+
+#include "benchmark/benchmark.h"
+
+namespace Envoy {
+namespace Stats {
+
+class StatsMatcherPerf {
+public:
+  StatsMatcherPerf() : pool_(symbol_table_) {}
+
+  void initMatcher() {
+    stats_matcher_impl_ = std::make_unique<StatsMatcherImpl>(stats_config_, symbol_table_);
+  }
+
+  envoy::type::matcher::v3::StringMatcher* inclusionList() {
+    return stats_config_.mutable_stats_matcher()->mutable_inclusion_list()->add_patterns();
+  }
+  envoy::type::matcher::v3::StringMatcher* exclusionList() {
+    return stats_config_.mutable_stats_matcher()->mutable_exclusion_list()->add_patterns();
+  }
+
+  SymbolTableImpl symbol_table_;
+  StatNamePool pool_;
+  std::unique_ptr<StatsMatcherImpl> stats_matcher_impl_;
+
+private:
+  envoy::config::metrics::v3::StatsConfig stats_config_;
+};
+
+} // namespace Stats
+} // namespace Envoy
+
+static void BM_Inclusion(benchmark::State& state) {
+  // Create matcher.
+  Envoy::Stats::StatsMatcherPerf context;
+  context.inclusionList()->set_suffix("1");
+  context.inclusionList()->set_prefix("2.");
+  context.initMatcher();
+
+  for (auto _ : state) {
+    for (auto idx = 0; idx < 1000; ++idx) {
+      Envoy::Stats::StatName stat_name = context.pool_.add(absl::StrCat("2.", idx % 10));
```

Comment:
Suggest pre-allocating the 10 stat_names in an array and doing a lookup, rather than having the stat-name pool addition measured as part of the benchmark.